lib/kms/wrapper.js

```diff
@@ -355,26 +357,49 @@ class KMS {
     }
 
     static checkHealth(log, cb) {
+        logger.debug('current KMS cache state', { lastChecked: cache.getLastChecked(), result: cache.getResult() });
         if (!client.healthcheck) {
             return cb(null, {
                 [implName]: { code: 200, message: 'OK' },
             });
         }
-        return client.healthcheck(log, err => {
-            const respBody = {};
-            if (err) {
-                respBody[implName] = {
-                    error: err.description,
-                    code: err.code,
-                };
-            } else {
-                respBody[implName] = {
-                    code: 200,
-                    message: 'OK',
-                };
-            }
-            return cb(null, respBody);
-        });
+
+        const shouldRefreshCache = cache.shouldRefresh();
+
+        // healthcheck only if cache needs refreshing
+        if (shouldRefreshCache) {
+            logger.debug('health check for KMS backend');
+            return client.healthcheck(log, (err) => {
+                const respBody = {};
+                if (err) {
+                    // Returning HTTP status code 200 allows CloudServer to start up and listen on port 8000
+                    // when KMS is unavailable, and ensures that the S3 frontend deep healthcheck
+                    // still allow requests to this CloudServer instance. This way, an unavailable KMIP
+                    // host or a connection issue with KMIP will not prevent the normal operation of CloudServer.
+                    respBody[implName] = {
+                        code: 200,
+                        message: 'KMS health check failed',
+                        description: err.description,
+                    };
+                    logger.warn('KMS health check failed', { errorCode: err.code, error: err.description });
+                } else {
+                    respBody[implName] = {
```

Comment:
I wonder if it could be interesting to put the date of the last cached health check if cached for a long time, so you can see it directly in the response without needing debug logs